src/components/Table/GridTable.tsx

```diff
@@ -777,36 +758,44 @@ export function filterRows<R extends Kinded>(
   columns: GridColumnWithId<R>[],
   rows: GridDataRow<R>[],
   filter: string | undefined,
-): ParentChildrenTuple<R>[] {
+): [string[], ParentChildrenTuple<R>[]] {
```

Comment:
I changed this method to also calc/return the flat `filteredRowIds` list while we're at it, otherwise we were re-building `filteredRowIds` while doing the `visibleDataRows` calc.